src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Tree/Constant.cs

```diff
@@ -26,22 +26,21 @@ public long Int64Value
         {
             get
             {
-                switch (Type.fundType())
+                switch (Type.FundamentalType)
                 {
                     case FUNDTYPE.FT_I8:
                     case FUNDTYPE.FT_U8:
                         return Val.Int64Val;
+
                     case FUNDTYPE.FT_U4:
                         return Val.UInt32Val;
-                    case FUNDTYPE.FT_I1:
-                    case FUNDTYPE.FT_I2:
-                    case FUNDTYPE.FT_I4:
-                    case FUNDTYPE.FT_U1:
-                    case FUNDTYPE.FT_U2:
-                        return Val.Int32Val;
+
                     default:
-                        Debug.Assert(false, "Bad fundType in getI64Value");
-                        return 0;
+                        Debug.Assert(
+                            Type.FundamentalType == FUNDTYPE.FT_I1 || Type.FundamentalType == FUNDTYPE.FT_I2
+                            || Type.FundamentalType == FUNDTYPE.FT_I4 || Type.FundamentalType == FUNDTYPE.FT_U1
+                            || Type.FundamentalType == FUNDTYPE.FT_U2, "Bad fundType in getI64Value");
+                        return Val.Int32Val;
                 }
```

Comment:
Are we sure we want to do this? are there other checks that might be affected by the `0` value returned in erroneous cases?

Reply:
Zero would be no better than any other value if something did hit it in error (though `Int32Val` would be zero in such a case anyway), but all calls are within branches that depend on the `FundamentalType` anyway, so the (non-debug) number of branches may as well be reduced.